geticeccversion.sh

```diff
@@ -46,26 +47,40 @@ MAC_ENV_PATH="${ICECC_ENV_DIR}/${ENV_NAME}"
 LINUX_ENV_PATH="${ICECC_LINUX_ENV_DIR}/${ENV_NAME}"
 
 if [ ! -e "$MAC_ENV_PATH" ]; then
-  CLANG_PATH="${CHROMIUM_PATH}/third_party/llvm-build/Release+Asserts/bin/clang"
+  LLVM_PATH="${CHROMIUM_PATH}/third_party/llvm-build/Release+Asserts"
+  CLANG_PATH="${LLVM_PATH}/bin/clang"
+  PLUGINS_PATH="${LLVM_PATH}/lib"
 
   if [ ! -x "$CLANG_PATH" ]; then
     echo "Error: Can't find clang executable at $CLANG_PATH." >&2
     exit 1
   fi
 
-  TEMP_ENV_FILENAME=`(cd $ICECC_ENV_DIR && exec 5>&1 && $ICECC_CREATE_ENV --clang $CLANG_PATH 1>/dev/null)`
+  mkdir -p $ICECC_ENV_TEMP_DIR
+  rm -rf {$ICECC_ENV_TEMP_DIR}/*
+
+  TEMP_ENV_FILENAME=`(cd $ICECC_ENV_TEMP_DIR && exec 5>&1 && $ICECC_CREATE_ENV --clang $CLANG_PATH 1>/dev/null)`
   if [ -z "$TEMP_ENV_FILENAME" ]; then
     echo "Error: couldn't get file name of generated file." >&2
     exit 1
   fi
 
-  TEMP_ENV_PATH=${ICECC_ENV_DIR}/${TEMP_ENV_FILENAME}
+  TEMP_ENV_PATH=${ICECC_ENV_TEMP_DIR}/${TEMP_ENV_FILENAME}
   if [ ! -e "$TEMP_ENV_PATH" ]; then
     echo "Error: Can't find environment created at $TEMP_ENV_PATH." >&2
     exit 1
   fi
 
-  mv "$TEMP_ENV_PATH" "$MAC_ENV_PATH"
+  pushd $ICECC_ENV_TEMP_DIR
+  tar xzf $TEMP_ENV_FILENAME 1>/dev/null
```

Comment:
There's no need to untar and retar the whole file - as long as it's not compressed you can add files to the archive using the 'r' option. Then you won't need a temporary directory to copy the files to before adding them to the archive.